### DISCUSSION TITLE
API returning empty array

### DISCUSSION THREAD

**Discussion by northrn** [supabase/supabase#6093] [Questions] [answered]
I am using Supabase in a NextJS project and I don't know why, but I am unable to query data from my database.

I am trying to query the "Product" table and when I console log out the result of getStaticProps, I just get an empty array.

I would appreciate any pointers!

Here is my index.js file:

``` javascript
import Head from 'next/head'
import Link from 'next/link'
import { fromImageToUrl, API_URL } from '../utils/urls';
import { supabase } from '../utils/supabaseClient';



export default function Home ({ product }) {
  
console.log(product)
return (
  <div>
    <Head>
      <title>Create Next App</title>
      <link rel="icon" href="/favicon.ico" class="bg-slate-50" />
    </Head>

    <div className="bg-white">
    <div className="max-w-2xl mx-auto py-16 px-4 sm:py-24 sm:px-6 lg:max-w-7xl lg:px-8">
      <h2 className="text-2xl font-extrabold tracking-tight text-gray-900 bg">Products</h2>

      <div className="mt-6 grid grid-cols-1 gap-y-10 gap-x-6 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8">
        {product.map((product) => (
          <div key={product.id} className="group relative">
            <Link href={`/products/${product.href}`}>
              <div>
                <div className="w-full min-h-80 bg-gray-200 aspect-w-1 aspect-h-1 rounded-md overflow-hidden group-hover:opacity-75 lg:h-80 lg:aspect-none">
                  <img
                    src={fromImageToUrl(product.imageSrc)}
                    alt={product.imageAlt}
                    className="w-full h-full object-center object-cover lg:w-full lg:h-full"
                  />
                </div>
                <div className="mt-4 flex justify-between">
                  <div>
                    <h3 className="text-sm text-gray-700">
                      <a href={product.href}>
                        <span aria-hidden="true" className="absolute inset-0" />
                        {product.name}
                      </a>
                    </h3>
                    <p className="mt-1 text-sm text-gray-500">{product.color}</p>
                  </div>
                  <p className="text-sm font-medium text-gray-900">{product.price}</p>
                </div>
              </div>
            </Link>
          </div>
        ))}
      </div>
    </div>
  </div>
 </div> 
)
        }


export async function getStaticProps() {
  const { data: product } = await supabase.from('product').select('*')
  return {
      props: {
          product 
      }
  }
}

**Comment by GaryAustin1** [comment, marked as answer]
Empty array usually means you don't meet RLS if you have it on.  This could be bad policy, or you don't actually have a user signed or appropriate key set at the time you make the call.  
getStaticProps runs from the server side to prerender, so if you have RLS, you will have to deal with getting a user's jwt setup on the server side before making the call or not have RLS for that call. 
Try turning RLS off if you have it on as a quick test.  If that works, add a select policy (auth.role() = 'anon').  If that works then you are not getting the jwt header to the supabase call from your signin.

**Reply by northrn** [reply]
Yep you got it! Thanks so much. I spent a whole day trying to figure it out... 

I notice that my getStaticPaths function is resulting in a 404. Do you see any issues here?

Thanks again!

```Javascript
export async function getStaticPaths() {
  //Retieve all products
  const { data: product } = await supabase.from('product').select('id')

  const paths = product.map(({id}) => ({
    params: { 
      id : id.toString(), 
     }
}))

  return {
      paths,
      fallback: false //Tell NextJS to display a 404 if params don't exist
  }
}

**Comment by GaryAustin1** [comment]
I don't use Nextjs but I assume you are not getting a result back there either.  Same questions would apply as before and do you have a id field in products.  What did you find in the first case?

**Reply by northrn** [reply]
Turning off RLS solved the original issue, which I thought would have solved this one as well. There is indeed an ID field in products (It is an integer with a value of "1" in this case). I may be doing something wrong in the context of Next rather than Supabase.

**Reply by GaryAustin1** [reply]
Having {id} in map seems strange to me... would it not be .map((id)=>({.....

**Reply by northrn** [reply]
I figured it out... It had to do with the file structure. I also found it odd that it's {id} rather than simply id. I was following a tutorial and that's the way he does it in the video and it seems to work now.

**Comment by Richard-vinu** [comment]
thank you , finally i am getting the data from supabase table
